metrics_gather.sh

```diff
@@ -214,6 +214,40 @@ av1-rt)
   $($TIMERDEC $AOMDEC --codec=av1 $AOMDEC_OPTS -o $BASENAME.y4m $BASENAME.ivf)
   SIZE=$(stat -c %s $BASENAME.ivf)
   ;;
+av2 | av2-ai | av2-ra | av2-ld | av2-as)
+  case $CODEC in
+    av2-ai)
+      CTC_PROFILE_OPTS="--cpu-used=0 --passes=1 --end-usage=q --cq-level=x --kf-min-dist=0 --kf-max-dist=0 --use-fixed-qp-offsets=1 --limit=30 --deltaq-mode=0 --enable-tpl-model=0 --enable-keyframe-filtering=0 --obu"
+      ;;
+    av2-ra)
+      CTC_PROFILE_OPTS="--cpu-used=0 --passes=1 --lag-in-frames=19 --auto-alt-ref=1 --min-gf-interval=16 --max-gf-interval=16 --gf-min-pyr-height=4 --gf-max-pyr-height=4 --limit=130 --kf-min-dist=65 --kf-max-dist=65 --use-fixed-qp-offsets=1 --deltaq-mode=0 --enable-tpl-model=0 --end-usage=q --enable-keyframe-filtering=0 --obu"
+      ;;
+    av2-ld)
+      CTC_PROFILE_OPTS="--cpu-used=0 --passes=1 --lag-in-frames=0 --min-gf-interval=16 --max-gf-interval=16 --gf-min-pyr-height=4 --gf-max-pyr-height=4 --limit=130 --kf-min-dist=9999 --kf-max-dist=9999  --use-fixed-qp-offsets=1 --deltaq-mode=0 --enable-tpl-model=0 --end-usage=q --subgop-config-str=ld --enable-keyframe-filtering=0 --obu"
+      ;;
+    av2-as)
+      CTC_PROFILE_OPTS="--cpu-used=0 --passes=1 --lag-in-frames=19 --auto-alt-ref=1 --min-gf-interval=16 --max-gf-interval=16 --gf-min-pyr-height=4 --gf-max-pyr-height=4 --limit=130 --kf-min-dist=65 --kf-max-dist=65 --use-fixed-qp-offsets=1 --deltaq-mode=0 --enable-tpl-model=0 --end-usage=q --enable-keyframe-filtering=0 --obu"
+      ;;
+    av2)
+      # generic, not currently used
+      CTC_PROFILE_OPTS=""
+      ;;
+  esac
+  case $(basename $FILE) in
+    BoxingPractice_3840x2160_5994fps_10bit_420.y4m | Crosswalk_3840x2160_5994fps_10bit_420.y4m | FoodMarket2_3840x2160_5994fps_10bit_420.y4m | Neon1224_3840x2160_2997fps.y4m | NocturneDance_3840x2160p_10bit_60fps.y4m | PierSeaSide_3840x2160_2997fps_10bit_420.y4m | Tango_3840x2160_5994fps_10bit_420.y4m | TimeLapse_3840x2160_5994fps_10bit_420.y4m)
```

Comment:
Since we already have `WIDTH` and `HEIGHT` variables parsed from the Y4M header I think it would be better to use those to determine the threading opts instead of file names. Also, maybe this is something we can consider doing for other encoders too.

Reply:
That is what I initially wrote, but the AV2 CTC specifically requires the a1 set to have different threading options than the rest, and other sets do have 4K videos.

In either case this is kind of a not scalable way to implement this, I could do something like a name check on the test set at a higher level and inject these into EXTRA_OPTS instead.

Reply:
I see, in that case I would prefer setting the set-specific behavior in AWCY, because I think of rd_tool as a generic tool to perform the encodes and output metrics as instructed without business logic embedded.

Reply:
I might not be per-set behavior exactly because we might have combined sets (e.g. run a1, a2, etc at once). I'm not sure how I want to do the combined sets yet.

Reply:
I have updated it to just not include this for now, you have to specify it manually. I will put it in, in an automatic way in a follow up.